### ISSUE TITLE
Vagrant up fails with Ubuntu 22.04

### ISSUE THREAD

**Issue by timotheenicolas** [hashicorp/vagrant#12751]
Vagrant fails to run VMs with Ubuntu 22.04 as host Operating System.
It looks like Ubuntu 22.04 uses OpenSSL 3.x and Vagrant net-ssh is not compatible yet.

### Vagrant version

Vagrant 2.2.19

### Host operating system

Ubuntu Jammy 22.04 LTS
with OpenSSL 3.0.2

### Guest operating system

Ubuntu Focal 20.04 LTS

### Vagrantfile

```ruby
Vagrant.configure(2) do |config|
  config.vm.box = "ubuntu/focal64"
end
```

### Debug output

```
Bringing machine 'default' up with 'virtualbox' provider...
==> default: Importing base box 'ubuntu/focal64'...
==> default: Matching MAC address for NAT networking...
==> default: Checking if box 'ubuntu/focal64' version '20220419.0.0' is up to date...
==> default: Setting the name of the VM: tmp_default_1650901938432_47703
==> default: Clearing any previously set network interfaces...
==> default: Preparing network interfaces based on configuration...
    default: Adapter 1: nat
==> default: Forwarding ports...
    default: 22 (guest) => 2222 (host) (adapter 1)
==> default: Running 'pre-boot' VM customizations...
==> default: Booting VM...
==> default: Waiting for machine to boot. This may take a few minutes...
    default: SSH address: 127.0.0.1:2222
    default: SSH username: vagrant
    default: SSH auth method: private key
==> default: Attempting graceful shutdown of VM...
==> default: Attempting graceful shutdown of VM...
==> default: Attempting graceful shutdown of VM...
==> default: Attempting graceful shutdown of VM...
==> default: Attempting graceful shutdown of VM...
==> default: Attempting graceful shutdown of VM...
==> default: Attempting graceful shutdown of VM...
==> default: Attempting graceful shutdown of VM...
==> default: Attempting graceful shutdown of VM...
==> default: Attempting graceful shutdown of VM...
==> default: Attempting graceful shutdown of VM...
==> default: Attempting graceful shutdown of VM...
==> default: Attempting graceful shutdown of VM...
==> default: Attempting graceful shutdown of VM...
==> default: Attempting graceful shutdown of VM...
==> default: Attempting graceful shutdown of VM...
==> default: Attempting graceful shutdown of VM...
/usr/share/rubygems-integration/all/gems/net-ssh-6.1.0/lib/net/ssh/transport/kex/ecdh_sha2_nistp256.rb:21:in `generate_key!': pkeys are immutable on OpenSSL 3.0 (OpenSSL::PKey::PKeyError)
        from /usr/share/rubygems-integration/all/gems/net-ssh-6.1.0/lib/net/ssh/transport/kex/ecdh_sha2_nistp256.rb:21:in `generate_key'
        from /usr/share/rubygems-integration/all/gems/net-ssh-6.1.0/lib/net/ssh/transport/kex/abstract.rb:32:in `initialize'
        from /usr/share/rubygems-integration/all/gems/net-ssh-6.1.0/lib/net/ssh/transport/algorithms.rb:437:in `new'
        from /usr/share/rubygems-integration/all/gems/net-ssh-6.1.0/lib/net/ssh/transport/algorithms.rb:437:in `exchange_keys'
        from /usr/share/rubygems-integration/all/gems/net-ssh-6.1.0/lib/net/ssh/transport/algorithms.rb:245:in `proceed!'
        from /usr/share/rubygems-integration/all/gems/net-ssh-6.1.0/lib/net/ssh/transport/algorithms.rb:184:in `accept_kexinit'
        from /usr/share/rubygems-integration/all/gems/net-ssh-6.1.0/lib/net/ssh/transport/session.rb:210:in `block in poll_message'
        from /usr/share/rubygems-integration/all/gems/net-ssh-6.1.0/lib/net/ssh/transport/session.rb:190:in `loop'
        from /usr/share/rubygems-integration/all/gems/net-ssh-6.1.0/lib/net/ssh/transport/session.rb:190:in `poll_message'
        from /usr/share/rubygems-integration/all/gems/net-ssh-6.1.0/lib/net/ssh/transport/session.rb:225:in `block in wait'
        from /usr/share/rubygems-integration/all/gems/net-ssh-6.1.0/lib/net/ssh/transport/session.rb:223:in `loop'
        from /usr/share/rubygems-integration/all/gems/net-ssh-6.1.0/lib/net/ssh/transport/session.rb:223:in `wait'
        from /usr/share/rubygems-integration/all/gems/net-ssh-6.1.0/lib/net/ssh/transport/session.rb:90:in `initialize'
        from /usr/share/rubygems-integration/all/gems/net-ssh-6.1.0/lib/net/ssh.rb:251:in `new'
        from /usr/share/rubygems-integration/all/gems/net-ssh-6.1.0/lib/net/ssh.rb:251:in `start'
        from /usr/share/rubygems-integration/all/gems/vagrant-2.2.19/plugins/communicators/ssh/communicator.rb:467:in `block (2 levels) in connect'
        from /usr/lib/ruby/3.0.0/timeout.rb:97:in `block in timeout'
        from /usr/lib/ruby/3.0.0/timeout.rb:35:in `block in catch'
        from /usr/lib/ruby/3.0.0/timeout.rb:35:in `catch'
        from /usr/lib/ruby/3.0.0/timeout.rb:35:in `catch'
        from /usr/lib/ruby/3.0.0/timeout.rb:112:in `timeout'
        from /usr/share/rubygems-integration/all/gems/vagrant-2.2.19/plugins/communicators/ssh/communicator.rb:433:in `block in connect'
        from /usr/share/rubygems-integration/all/gems/vagrant-2.2.19/lib/vagrant/util/retryable.rb:17:in `retryable'
        from /usr/share/rubygems-integration/all/gems/vagrant-2.2.19/plugins/communicators/ssh/communicator.rb:432:in `connect'
        from /usr/share/rubygems-integration/all/gems/vagrant-2.2.19/plugins/communicators/ssh/communicator.rb:149:in `ready?'
        from /usr/share/rubygems-integration/all/gems/vagrant-2.2.19/lib/vagrant/machine.rb:281:in `guest'
        from /usr/share/rubygems-integration/all/gems/vagrant-2.2.19/lib/vagrant/action/builtin/graceful_halt.rb:50:in `call'
        from /usr/share/rubygems-integration/all/gems/vagrant-2.2.19/lib/vagrant/action/warden.rb:48:in `call'
        from /usr/share/rubygems-integration/all/gems/vagrant-2.2.19/lib/vagrant/action/builder.rb:149:in `call'
        from /usr/share/rubygems-integration/all/gems/vagrant-2.2.19/lib/vagrant/action/runner.rb:89:in `block in run'
```

### Expected behavior

Vagrant should run the VM without issue.

### Actual behavior

Vagrant fails to run the VM.

### Steps to reproduce

1. vagrant up

### References

It seems related to the rubygem net-ssh that do not support OpenSSL 3.0
https://github.com/net-ssh/net-ssh/issues/843

**Comment by ashleykleynhans** [comment]
I also ran into this issue with Ubuntu 22.04, but your logs refer to `ubuntu/focal64` which is actually Ubuntu 20.04.

With ubuntu 22.04 (ubuntu/jammy64), `vagrant ssh` works perfectly, but `vagrant up` fails with multiple Authentication failures.

The workaround for me was to remove all keys from the SSH agent on the host machine:

```ssh-add -D```

And then `vagrant up` works perfectly.

**Comment by dimur** [comment]
> I also ran into this issue with Ubuntu 22.04, but your logs refer to `ubuntu/focal64` which is actually Ubuntu 20.04.
> 
> With ubuntu 22.04 (ubuntu/jammy64), `vagrant ssh` works perfectly, but `vagrant up` fails with multiple Authentication failures.
> 
> The workaround for me was to remove all keys from the SSH agent on the host machine:
> 
> `ssh-add -D`
> 
> And then `vagrant up` works perfectly.

Unfortunately, this workaround doesn't work for me (Ubuntu 22.04 (ubuntu/jammy64)) ((

**Comment by ashleykleynhans** [comment]
> Unfortunately, this workaround doesn't work for me (Ubuntu 22.04 (ubuntu/jammy64)) ((

Probably depends on the host Operating system, I should have mentioned I am using macOS Monterey 12.2.1.

**Comment by dimur** [comment]
> > Unfortunately, this workaround doesn't work for me (Ubuntu 22.04 (ubuntu/jammy64)) ((
> 
> Probably depends on the host Operating system, I should have mentioned I am using macOS Monterey 12.2.1.

My host Operating system is Ubuntu 22.04 :-) like the author has:
> Vagrant fails to run VMs with Ubuntu 22.04 as host Operating System.

And I try to run VMs with `ubuntu/xenial64` for instance.

**Comment by KarthickSudhakar** [comment]
I recently updated my operating system from Pop!_OS 21.10 to Pop!_OS 22.04. I am facing the same problem. Whenever I run "vagrant up" on any lab I have it is throwing me the following error.

  managed1: SSH auth method: private key
/usr/share/rubygems-integration/all/gems/net-ssh-6.1.0/lib/net/ssh/transport/kex/ecdh_sha2_nistp256.rb:21:in `generate_key!': pkeys are immutable on OpenSSL 3.0 (OpenSSL::PKey::PKeyError)
	from /usr/share/rubygems-integration/all/gems/net-ssh-6.1.0/lib/net/ssh/transport/kex/ecdh_sha2_nistp256.rb:21:in `generate_key'
	from /usr/share/rubygems-integration/all/gems/net-ssh-6.1.0/lib/net/ssh/transport/kex/abstract.rb:32:in `initialize'
	from /usr/share/rubygems-integration/all/gems/net-ssh-6.1.0/lib/net/ssh/transport/algorithms.rb:437:in `new'
	from /usr/share/rubygems-integration/all/gems/net-ssh-6.1.0/lib/net/ssh/transport/algorithms.rb:437:in `exchange_keys'
	from /usr/share/rubygems-integration/all/gems/net-ssh-6.1.0/lib/net/ssh/transport/algorithms.rb:245:in `proceed!'
	from /usr/share/rubygems-integration/all/gems/net-ssh-6.1.0/lib/net/ssh/transport/algorithms.rb:184:in `accept_kexinit'
	from /usr/share/rubygems-integration/all/gems/net-ssh-6.1.0/lib/net/ssh/transport/session.rb:210:in `block in poll_message'
	from /usr/share/rubygems-integration/all/gems/net-ssh-6.1.0/lib/net/ssh/transport/session.rb:190:in `loop'
	from /usr/share/rubygems-integration/all/gems/net-ssh-6.1.0/lib/net/ssh/transport/session.rb:190:in `poll_message'
	from /usr/share/rubygems-integration/all/gems/net-ssh-6.1.0/lib/net/ssh/transport/session.rb:225:in `block in wait'
	from /usr/share/rubygems-integration/all/gems/net-ssh-6.1.0/lib/net/ssh/transport/session.rb:223:in `loop'
	from /usr/share/rubygems-integration/all/gems/net-ssh-6.1.0/lib/net/ssh/transport/session.rb:223:in `wait'
	from /usr/share/rubygems-integration/all/gems/net-ssh-6.1.0/lib/net/ssh/transport/session.rb:90:in `initialize'
	from /usr/share/rubygems-integration/all/gems/net-ssh-6.1.0/lib/net/ssh.rb:251:in `new'
	from /usr/share/rubygems-integration/all/gems/net-ssh-6.1.0/lib/net/ssh.rb:251:in `start'
	from /usr/share/rubygems-integration/all/gems/vagrant-2.2.19/plugins/communicators/ssh/communicator.rb:467:in `block (2 levels) in connect'

**Comment by robert-lewon** [comment]
I confirm the same:
- host: Ubuntu 22.04
- guest(s): Debian 10, Ubuntu 18.04

Exception: `generate_key!': pkeys are immutable on OpenSSL 3.0 (OpenSSL::PKey::PKeyError)`

**Comment by nicolas-t** [comment]
A workaround that seems to work is to uninstall vagrant with
`sudo apt-get remove vagrant`
and to install a .deb file found at the bottom of this page : 
https://www.vagrantup.com/downloads

**Comment by linuxem** [comment]
> A workaround that seems to work is to uninstall vagrant with `sudo apt-get remove vagrant` and to install a .deb file found at the bottom of this page : https://www.vagrantup.com/downloads

work for me!

**Comment by timzwier** [comment]
> A workaround that seems to work is to uninstall vagrant with `sudo apt-get remove vagrant` and to install a .deb file found at the bottom of this page : https://www.vagrantup.com/downloads

Works for me (on Pop!_OS 22.04)

**Comment by bensummerhayes-ocsi** [comment]
Meh! 

> A workaround that seems to work is to uninstall vagrant with `sudo apt-get remove vagrant` and to install a .deb file found at the bottom of this page : https://www.vagrantup.com/downloads

Meh, didn't work for me. I execute "vagrant up" and get the following: "Exec error: fork/exec /opt/vagrant/embedded/bin/ruby: no such file or directory"

Kubuntu 22.04

Any suggestions would be greatly received as not having my VM working is a bit of a disaster

**Comment by nicolas-t** [comment]
@bensummerhayes-ocsi Are you sure you downloaded the correct .deb file ? Debian Amd worked for me.

**Comment by bensummerhayes-ocsi** [comment]
@nicolas-t 

Hello mate...thank you for reaching out. I just tried a slightly different approach by installing the package via the terminal other than via the UI Package Installer: 

`cd /Downloads`
`sudo dpkg -i vagrant_2.2.19_i686.deb` 

And I'm pleased to report it works!

Thank you to everyone here for taking the time to share your findings and fixes :)

**Comment by rubo77** [comment]
The workarounds didn’t work for me completely, I did them both: **install vagrant 2.2.19** and removed all keys from the SSH agent on the host machine:

    ssh-add -D

 Now the **error** messages are **still there**, but the virtual machine **starts OK** anyway! 
🤷

(also after upgrade from 21.10 to 22.04)

**Comment by knightSarai** [comment]
> A workaround that seems to work is to uninstall vagrant with `sudo apt-get remove vagrant` and to install a .deb file found at the bottom of this page : https://www.vagrantup.com/downloads
I can confirm this worked for me :)

**Comment by dvillaj** [comment]
> > A workaround that seems to work is to uninstall vagrant with `sudo apt-get remove vagrant` and to install a .deb file found at the bottom of this page : https://www.vagrantup.com/downloads
> 
> work for me!

Me too!!!

**Comment by rmarshall31** [comment]
For me the workaround will spin up a VM, but it won't run the provisioning steps after the VM is up.

**Comment by vinadoti** [comment]
> @nicolas-t
> 
> Olá amigo... obrigado por entrar em contato. de tentar uma abordagem um pouco diferente instalando o pacote por meio do terminal que não seja pelo UI Package Installer:
> 
> `cd /Downloads` `sudo dpkg -i vagrant_2.2.19_i686.deb`
> 
> E tenho o prazer de informar que funciona!
> 
> Obrigado a todos aqui por dedicarem tempo para compartilhar suas descobertas e um :)

I tried to do what you suggested but it didn't work, could you do a step by step to help?

**Comment by ashleykleynhans** [comment]
@vinadoti 

From the terminal:

First download the suitable installation file for your architecture type:

```wget https://releases.hashicorp.com/vagrant/2.2.19/vagrant_2.2.19_x86_64.deb```

or

```wget https://releases.hashicorp.com/vagrant/2.2.19/vagrant_2.2.19_i686.deb```

Then install it:

```sudo dpkg -i /vagrant_2.2.19_x86_64.deb```
or

```sudo dpkg -i vagrant_2.2.19_i686.deb```

**Comment by rubo77** [comment]
I can confirm, on Ubuntu 22.04, that manually installing 

    sudo apt-get remove vagrant
    # this will remove the packages vagrant and vagrant-libvirt
    sudo dpkg -i vagrant_2.2.19_i686.deb

This didn't work though:

```
curl -fsSL https://apt.releases.hashicorp.com/gpg | sudo apt-key add -
sudo apt-add-repository "deb [arch=amd64] https://apt.releases.hashicorp.com $(lsb_release -cs) main"
sudo apt-get update && sudo apt-get install vagrant
```

***REALLY STRANGE!!!***
maybe it is important to uninstall the package `vagrant-libvirt`?

**Comment by timotheenicolas** [comment]
`net-ssh` released a new beta version `7.0.0.beta1` that fix the SSH issue. https://github.com/net-ssh/net-ssh/pull/864
You can update the `net-ssh` gem as a workaround with `sudo gem install net-ssh 7.0.0.beta1` and `vagrant up` will works again.

**Comment by maurizi** [comment]
Quick correction to the above: that should be `sudo gem install net-ssh --version 7.0.0.beta1`

**Comment by daniram94** [comment]
> vagrant_2.2.19_i686.deb

it's works for me with ubuntu 22.04 and 84& 64 bits package. I get it from this url and install it with dpkg -i <package_name>
wget https://releases.hashicorp.com/vagrant/2.2.19/vagrant_2.2.19_x86_64.deb

**Comment by arnobnq** [comment]
The workaround doesn't work for me on Pop!_OS 22.04, vagrant doesn't find the libvirt provider after manually installing vagrant via the deb file, even after re-installing vagrant-libvirt from the repository.

**Comment by elreydetoda** [comment]
> The workaround doesn't work for me on Pop!_OS 22.04, vagrant doesn't find the libvirt provider after manually installing vagrant via the deb file, even after re-installing vagrant-libvirt from the repository.

I think I figured out the issue, posted in the forums about it here: https://discuss.hashicorp.com/t/vagrant-up-error-on-ubuntu-22-04-pkeys-are-immutable-on-openssl-3-0/38734/6?u=elreydetoda

I use `vagrant-libvirt` as well and it installed perfectly fine after doing what I did.

**Comment by arnobnq** [comment]
> > The workaround doesn't work for me on Pop!_OS 22.04, vagrant doesn't find the libvirt provider after manually installing vagrant via the deb file, even after re-installing vagrant-libvirt from the repository.
> 
> I think I figured out the issue, posted in the forums about it here: https://discuss.hashicorp.com/t/vagrant-up-error-on-ubuntu-22-04-pkeys-are-immutable-on-openssl-3-0/38734/6?u=elreydetoda
> 
> I use `vagrant-libvirt` as well and it installed perfectly fine after doing what I did. I also use the hashicorp repos for the installation (not a `.deb` file)
I tried your workaround, but I still get `No usable default provider could be found for your system` when issuing any vagrant command after forcing `apt install vagrant=2.2.19` (I can confirm the hashicorp repo is used here) then `apt install vagrant-libvirt`.

**Comment by maltewhiite** [comment]
> A workaround that seems to work is to uninstall vagrant with `sudo apt-get remove vagrant` and to install a .deb file found at the bottom of this page : https://www.vagrantup.com/downloads

Works for me

**Comment by elreydetoda** [comment]
> > > The workaround doesn't work for me on Pop!_OS 22.04, vagrant doesn't find the libvirt provider after manually installing vagrant via the deb file, even after re-installing vagrant-libvirt from the repository.
> > 
> > 
> > I think I figured out the issue, posted in the forums about it here: https://discuss.hashicorp.com/t/vagrant-up-error-on-ubuntu-22-04-pkeys-are-immutable-on-openssl-3-0/38734/6?u=elreydetoda
> > I use `vagrant-libvirt` as well and it installed perfectly fine after doing what I did. I also use the hashicorp repos for the installation (not a `.deb` file)
> 
> I tried your workaround, but I still get `No usable default provider could be found for your system` when issuing any vagrant command after forcing `apt install vagrant=2.2.19` (I can confirm the hashicorp repo is used here) then `apt install vagrant-libvirt`.

@arnobnq this sounds more like an issue with path'ing on your system and vagrant finding the path to your vagrant-libvirt plugin. Maybe we should move this to [vagrant's HashiCorp Discuss forums](https://discuss.hashicorp.com/c/vagrant/24)? If you want to, you can just reply to this comment with the link of the forum post you've made and I can try and answer there in more depth if the rest of this post doesn't help you. (don't want to keep notifying people if it's an unrelated issue, and if others have the same issue they can follow the link you'll post to the forum)

So...the only other thing that I can think of is that your Pop_OS! system still has some artifacts from your previous installation. Whenever you did the `apt remove vagrant` if you don't do `--purge --autoremove` (**NOTE:** this will remove all related installed packages with vagrant and might break other dependencies (to have less of a chance for that just do `--purge` when uninstalling vagrant)) as well it probably won't remove directories from the old install. So, it might be referencing an old plugin directory.

First I was able to tell that vagrant could find my plugin by running this: `vagrant plugin list` and it listed it:

```console
$ vagrant plugin list
vagrant-libvirt (0.7.0, system)
```

I was able to find my vagrant-libvirt plugin installed on my system by doing the following command:
```console
$ sudo find / -iname vagrant-libvirt 2>/dev/null
/home/<username>/.vagrant.d/gems/2.7.4/gems/vagrant-libvirt-0.7.0/lib/vagrant-libvirt
/run/libvirt/network/vagrant-libvirt
/usr/share/rubygems-integration/all/gems/vagrant-libvirt-0.7.0/lib/vagrant-libvirt
/usr/share/doc/vagrant-libvirt
```

So, I'd make sure that your plugin is installed in a similar location.

Alternatively, if none of this stuff helps and you're tired of not having your vagrant setup. You could also build the vagrant-libvirt plugin [based on their docs](https://github.com/vagrant-libvirt/vagrant-libvirt#installation)

**Comment by konstruktoid** [comment]
If unable to upgrade, you could try using:
```sh
jammy.vm.provision "shell",
    inline: "echo 'PubkeyAcceptedKeyTypes=+ssh-rsa' >> /etc/ssh/sshd_config"
```

See https://discourse.ubuntu.com/t/jammy-jellyfish-release-notes/24668: "ssh-rsa is now [disabled by default in OpenSSH](https://www.openssh.com/txt/release-8.8)"

**Comment by elreydetoda** [comment]
> If unable to upgrade, you could try using:
> ```sh
> jammy.vm.provision "shell",
>     inline: "echo 'PubkeyAcceptedKeyTypes=+ssh-rsa' >> /etc/ssh/sshd_config"
> ```
> 
> See https://discourse.ubuntu.com/t/jammy-jellyfish-release-notes/24668: "ssh-rsa is now [disabled by default in OpenSSH](https://www.openssh.com/txt/release-8.8)"

I'd recommend putting that in something like `/etc/ssh/sshd_config.d/extra_key_type.conf`

That way when upgrades come through you aren't modifying the sshd config directly, and having to deal with merging upstream with your current sshd config.

The `Include /etc/ssh/sshd_config.d/*.conf` line near the top of your `/etc/ssh/sshd_config` grabs all those config files for you automatically.

**Comment by arnobnq** [comment]
> > > > The workaround doesn't work for me on Pop!_OS 22.04, vagrant doesn't find the libvirt provider after manually installing vagrant via the deb file, even after re-installing vagrant-libvirt from the repository.
> > > 
> > > 
> > > I think I figured out the issue, posted in the forums about it here: https://discuss.hashicorp.com/t/vagrant-up-error-on-ubuntu-22-04-pkeys-are-immutable-on-openssl-3-0/38734/6?u=elreydetoda
> > > I use `vagrant-libvirt` as well and it installed perfectly fine after doing what I did. I also use the hashicorp repos for the installation (not a `.deb` file)
> > 
> > 
> > I tried your workaround, but I still get `No usable default provider could be found for your system` when issuing any vagrant command after forcing `apt install vagrant=2.2.19` (I can confirm the hashicorp repo is used here) then `apt install vagrant-libvirt`.
> 
> @arnobnq this sounds more like an issue with path'ing on your system and vagrant finding the path to your vagrant-libvirt plugin. Maybe we should move this to [vagrant's HashiCorp Discuss forums](https://discuss.hashicorp.com/c/vagrant/24)? If you want to, you can just reply to this comment with the link of the forum post you've made and I can try and answer there in more depth if the rest of this post doesn't help you. (don't want to keep notifying people if it's an unrelated issue, and if others have the same issue they can follow the link you'll post to the forum)
> 
> So...the only other thing that I can think of is that your Pop_OS! system still has some artifacts from your previous installation. Whenever you did the `apt remove vagrant` if you don't do `--purge --autoremove` (**NOTE:** this will remove all related installed packages with vagrant and might break other dependencies (to have less of a chance for that just do `--purge` when uninstalling vagrant)) as well it probably won't remove directories from the old install. So, it might be referencing an old plugin directory.
> 
> First I was able to tell that vagrant could find my plugin by running this: `vagrant plugin list` and it listed it:
> 
> ```
> $ vagrant plugin list
> vagrant-libvirt (0.7.0, system)
> ```
> 
> I was able to find my vagrant-libvirt plugin installed on my system by doing the following command:
> 
> ```
> $ sudo find / -iname vagrant-libvirt 2>/dev/null
> /home/<username>/.vagrant.d/gems/2.7.4/gems/vagrant-libvirt-0.7.0/lib/vagrant-libvirt
> /run/libvirt/network/vagrant-libvirt
> /usr/share/rubygems-integration/all/gems/vagrant-libvirt-0.7.0/lib/vagrant-libvirt
> /usr/share/doc/vagrant-libvirt
> ```
> 
> So, I'd make sure that your plugin is installed in a similar location.
> 
> Alternatively, if none of this stuff helps and you're tired of not having your vagrant setup. You could also build the vagrant-libvirt plugin [based on their docs](https://github.com/vagrant-libvirt/vagrant-libvirt#installation)

Thanks for these tips. I first tried uninstalling vagrant with `--purge` and `--autoremove` and reinstalling both packages, which did not seem to help. I then proceeded to build the module as you suggested, which was surprisingly simple and actually solved the problem. This is what I did : 
```
# as root
apt-get remove --purge vagrant-libvirt
apt-get build-dep vagrant ruby-libvirt
apt-get install libxslt-dev libxml2-dev libvirt-dev zlib1g-dev ruby-dev
apt-get install libguestfs-tools

# as my user
vagrant plugin install vagrant-libvirt
```
Vagrant successfully located this newly installed version, and everything works out :

```
$ vagrant plugin list
vagrant-libvirt (0.8.2, global)
```

The vagrant-libvirt package installed the module at `/usr/share/rubygems-integration/all/gems/vagrant-libvirt-0.7.0/lib/vagrant-libvirt`, whereas installing via `vagrant plugin install` puts files in my user directory `/home/<user>/.vagrant.d/gems/2.7.4/gems/vagrant-libvirt-0.8.2/lib/vagrant-libvirt`.

**Comment by JasonYangShadow** [comment]
> Quick correction to the above: that should be `sudo gem install net-ssh --version 7.0.0.beta1`

this one works for me

**Comment by RafalSkolasinski** [comment]
Happening on Ubuntu 22.04 trying to run ubuntu/jammy64 using vagrant from apt-get.

**Comment by gillmoreno** [comment]
> A workaround that seems to work is to uninstall vagrant with `sudo apt-get remove vagrant` and to install a .deb file found at the bottom of this page : https://www.vagrantup.com/downloads

Worked for me too!

**Comment by lucaskanashiro** [comment]
FWIW net-ssh was fixed in Ubuntu 22.04, and therefore vagrant is working again. All you need to do is to update your systems.

**Comment by mikzuit** [comment]
> FWIW net-ssh was fixed in Ubuntu 22.04, and therefore vagrant is working again. All you need to do is to update your systems.

Confirmed, should be marked to "closed" @timotheenicolas
